kafka/conn.py

```diff
@@ -573,7 +573,7 @@ def _handle_sasl_handshake_response(self, future, response):
         elif self.config['sasl_mechanism'].startswith("SCRAM-SHA-"):
             return self._try_authenticate_scram(future)
         elif self.config['sasl_mechanism'] == 'AWS_MSK_IAM':
-            return self._try_authenticate_aws_msk_iam(future, self.config["sasl_aws_msk_iam_session"])
+            return self._try_authenticate_aws_msk_iam(future, self.config["sasl_aws_msk_iam_role_arn"])
```

Comment:
config is initialized in __init__, so there should be no need to pass it as a parameter.